### PR TITLE
2.15 merge

### PR DESCRIPTION
**Pull request by mateusz-sekara**
Merging chainlink repository from `v2.15.0`

### CODE REVIEW

File: contracts/src/v0.8/keystone/KeystoneFeedsConsumer.sol [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
-
-contract KeystoneFeedsConsumer is IReceiver, KeystoneFeedsPermissionHandler {
-  using KeystoneFeedDefaultMetadataLib for bytes;
 
+contract KeystoneFeedsConsumer is IReceiver, OwnerIsCreator, IERC165 {
   event FeedReceived(bytes32 indexed feedId, uint224 price, uint32 timestamp);
 
+  error UnauthorizedSender(address sender);
+  error UnauthorizedWorkflowOwner(address workflowOwner);
+  error UnauthorizedWorkflowName(bytes10 workflowName);
```
Comment (makramkd) [inline contracts/src/v0.8/keystone/KeystoneFeedsConsumer.sol:13]: @0xsuryansh can you check that this is not a regression?

File: core/services/chainlink/application.go [hunk 2]
```diff
@@ -201,29 +199,14 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
...
-			srvcs = append(srvcs, remoteDispatcher)
-
-			dispatcher = remoteDispatcher
-		} else {
-			dispatcher = opts.CapabilitiesDispatcher
-			externalPeerWrapper = opts.CapabilitiesPeerWrapper
-		}
-
-		srvcs = append(srvcs, externalPeerWrapper)
+	var capabilityRegistrySyncer registrysyncer.Syncer
```
Comment (asoliman92) [inline core/services/chainlink/application.go]: Code here is a bit outdated. even if 2.15 is 2 weeks ago I think it was different as most of the part related to the capabilities was last edited first week of August. Can you double check nothing was auto resolved wrongly here?

File: .mockery.yaml [hunk 3]
```diff
@@ -258,14 +258,10 @@ packages:
       ORM:
       Runner:
       PipelineParamUnmarshaler:
-  github.com/smartcontractkit/chainlink/v2/core/services/headreporter:
+  github.com/smartcontractkit/chainlink/v2/core/services/promreporter:
```
Comment (asoliman92) [inline .mockery.yaml:261]: I'm not familiar with prom or headreporter. What's the reason for removing headreporter completely like this?

Comment (mateusz-sekara) [reply .mockery.yaml:261]: That's the state of the repo on v2.15.0. It seems that promreporter replaced headreporter

File: core/services/chainlink/application.go (hunk 2 above)
Comment (mateusz-sekara) [reply core/services/chainlink/application.go]: 2.15 was cut on 30th July from develop, which means that code is at least a month behind the current develop

**Review by asoliman92 — APPROVED** [review]

**Review by asoliman92 — APPROVED** [review]

**Comment by mateusz-sekara** [comment]
Protobuf checks fails because of this breaking change https://github.com/smartcontractkit/chainlink/pull/13750

File: .swp
Comment (makramkd) [inline .swp:1]: remove

**Review by makramkd — CHANGES_REQUESTED** [review]

**Review by makramkd — APPROVED** [review]

**Review by RensR — APPROVED** [review]

**Pull request merged**